### PR TITLE
Use the same ReturnTypeExtension for array_merge() and array_replace()

### PR DESCRIPTION
**Pull request by staabm**
looking into `array_replace` right after working on `array_merge` makes me think, those 2 functions are pretty the same, especially when you only think about the types involved. for sure at runtime they do things differently, but type-wise is it even identical?

googling a bit seems to approve my guess: https://stackoverflow.com/a/34367698

the test-suite also agrees, so lets drop the unnecessary code

### CODE REVIEW

File: tests/PHPStan/Analyser/data/array-replace.php [hunk 1]
```diff
@@ -23,9 +23,9 @@ public function arrayReplace($array1, $array2): void
 	 */
 	public function arrayReplaceArrayShapes($array1, $array2): void
 	{
-		assertType("non-empty-array<'bar'|'foo', '1'|'2'>", array_replace($array1));
-		assertType("non-empty-array<'bar'|'foo', '1'|'2'>", array_replace([], $array1));
-		assertType("non-empty-array<'bar'|'foo', '1'|'2'|'4'>", array_replace($array1, $array2));
+		assertType("array{foo: '1', bar: '2'}", array_replace($array1));
+		assertType("array{foo: '1', bar: '2'}", array_replace([], $array1));
+		assertType("array{foo: '1', bar: '4'}", array_replace($array1, $array2));
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/array-replace.php:28]: here we are getting more specific types now, .. the same stuff, we recently added to array_merge

**Comment by staabm** [comment]
looked into the 3 failling jobs.. errors seem unrelated to `array_replace`

**Comment by ondrejmirtes** [comment]
When in doubt, check what's currently failing on the default branch in https://github.com/phpstan/phpstan :) I'm working on it.

**Comment by ondrejmirtes** [comment]
It's interesting that the test suite passes but this immediately raises a red flag for me. Surely in some cases the functions will have different results.

Take the example from your link: https://3v4l.org/FZdeA, https://phpstan.org/r/49f0b92b-8e36-4942-b1af-bbe368a1d896

PHPStan has wrong result for array_merge and slightly imprecise result for array_replace.

Calling these functions for two constant arrays isn' going to be very common but these bugs can surely be demonstrated on other cases as well.

These functions aren't aliases of each other, the extension have to be kept separate. Common parts can be reused through a helper.

Feel free to send a PR that fixes the behaviour, thanks.

**Pull request closed without merging**